make client fails

I attempted to run make client against tip to check it was working before updating the schema but it appears to be broken:

```
liam@figroll:~/branches/python-libjuju$ make client
.tox/py3/bin/python -m juju.client.facade -s "juju/client/schemas*" -o juju/client/
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 810, in <module>
    main()
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 801, in main
    captures = generate_facades(options)
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 767, in generate_facades
    buildTypes(schema, captures[schema.version])
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 281, in buildTypes
    textwrap.indent(args.get_doc(), INDENT * 2))]
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 253, in get_doc
    return self._get_arg_str(True, "\n")
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 232, in _get_arg_str
    parts.append(self._format(item[0], item[1], typed))
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 223, in _format
    strcast(rtype)
  File "/home/liam/branches/python-libjuju/juju/client/facade.py", line 174, in strcast
    if issubclass(kind, typing.GenericMeta):
TypeError: issubclass() arg 1 must be a class
Makefile:20: recipe for target 'client' failed
make: *** [client] Error 1
```

--- a/juju/client/facade.py
+++ b/juju/client/facade.py
@@ -171,13 +171,13 @@ def name_to_py(name):
 
 
 def strcast(kind, keep_builtins=False):
-    if issubclass(kind, typing.GenericMeta):
-        return str(kind)[1:]
-    if str(kind).startswith('~'):
-        return str(kind)[1:]
     if (kind in basic_types or
             type(kind) in basic_types) and keep_builtins is False:
         return kind.__name__
+    if str(kind).startswith('~'):
+        return str(kind)[1:]
+    if issubclass(kind, typing.GenericMeta):
+        return str(kind)[1:]
     return kind
 
 
@@ -291,6 +291,13 @@ class {}(Type):
                     source.append("{}self.{} = {}".format(INDENT * 2,
                                                           arg_name,
                                                           arg_name))
+                elif type(arg_type) is typing.TypeVar:
+                    source.append("{}self.{} = {}.from_json({}) "
+                                  "if {} else None".format(INDENT * 2,
+                                                           arg_name,
+                                                           arg_type_name,
+                                                           arg_name,
+                                                           arg_name))
                 elif issubclass(arg_type, typing.Sequence):
                     value_type = (
                         arg_type_name.__parameters__[0]
@@ -326,13 +333,6 @@ class {}(Type):
                         source.append("{}self.{} = {}".format(INDENT * 2,
                                                               arg_name,
                                                               arg_name))
-                elif type(arg_type) is typing.TypeVar:
-                    source.append("{}self.{} = {}.from_json({}) "
-                                  "if {} else None".format(INDENT * 2,
-                                                           arg_name,
-                                                           arg_type_name,
-                                                           arg_name,
-                                                           arg_name))
                 else:
                     source.append("{}self.{} = {}".format(INDENT * 2,
                                                           arg_name,

